common/src/main/java/com/wynntils/utils/wynn/ContainerUtils.java

```diff
@@ -66,7 +66,31 @@ public static void clickOnSlot(int clickedSlot, int containerId, int mouseButton
                 changedSlots));
     }
 
+    public static void shiftClickOnSlot(int clickedSlot, int containerId, int mouseButton, List<ItemStack> items) {
+        Int2ObjectMap<ItemStack> changedSlots = new Int2ObjectOpenHashMap<>();
+        changedSlots.put(clickedSlot, new ItemStack(Items.AIR));
+
+        int transactionId = 0;
+
+        McUtils.sendPacket(new ServerboundContainerClickPacket(
+                containerId,
+                transactionId,
+                clickedSlot,
+                mouseButton,
+                ClickType.QUICK_MOVE,
+                items.get(clickedSlot),
+                changedSlots));
+    }
+
     public static void closeContainer(int containerId) {
         McUtils.sendPacket(new ServerboundContainerClosePacket(containerId));
     }
+
+    /**
+     * Closes invisible containers opened in the background, without closing the visible screen.
+     */
+    public static void closeBackgroundContainer() {
```

Comment:
This sounds scary...

Reply:
Can't use queries without it, unless we remove the open container check. It's still "opened" in the background when our custom screen is on top

Reply:
I'm too tired to fully understand this now. But if @kristofbolyai says it is fine, I guess I am too.

Reply:
I have a round of reviewing specifically for the query system changes to do, I'll make sure to check if this is the best we can do.